pkg/api/server/v1alpha2/results_test.go

```diff
@@ -278,6 +279,43 @@ func TestDeleteResult(t *testing.T) {
 	})
 }
 
+func TestCascadeDelete(t *testing.T) {
+	srv, err := New(test.NewDB(t))
+	if err != nil {
+		t.Fatalf("failed to create server: %v", err)
+	}
+
+	ctx := context.Background()
+	result, err := srv.CreateResult(ctx, &pb.CreateResultRequest{
+		Parent: "foo",
+		Result: &pb.Result{
+			Name: "foo/results/bar",
+		},
+	})
+	if err != nil {
+		t.Fatalf("CreateResult: %v", err)
+	}
+	r, err := srv.CreateRecord(ctx, &pb.CreateRecordRequest{
+		Parent: result.GetName(),
+		Record: &pb.Record{
+			Name: recordutil.FormatName(result.GetName(), "baz"),
+		},
+	})
+	if err != nil {
+		t.Fatalf("CreateRecord(): %v", err)
+	}
+
+	t.Run("success", func(t *testing.T) {
+		if _, err = srv.DeleteResult(ctx, &pb.DeleteResultRequest{Name: result.GetName()}); err != nil {
+			t.Fatalf("could not delete the result: %v", err)
+		}
+		_, err = srv.GetRecord(ctx, &pb.GetRecordRequest{Name: r.GetName()})
+		if status.Code(err) != codes.NotFound {
+			t.Fatalf("cascade delete failed: %v", err)
```

Comment:
nit: scoping + additional flavor text on error
```suggestion
		if got, err := srv.GetRecord(ctx, &pb.GetRecordRequest{Name: r.GetName()}); status.Code(err) != codes.NotFound {
			t.Fatalf("cascade delete failed - expected Record to be deleted, got: (%+v, %v)", got, err)
```